storage/storage-client/src/lib.rs

```diff
@@ -41,24 +42,24 @@ use storage_proto::{
 
 /// This provides storage read interfaces backed by real storage service.
 pub struct StorageReadServiceClient {
-    addr: String,
+    http_addr: String,
     client: Mutex<Option<StorageClient<tonic::transport::Channel>>>,
 }
 
 impl StorageReadServiceClient {
-    /// Constructs a `StorageReadServiceClient` with given host and port.
-    pub fn new(host: &str, port: u16) -> Self {
-        let addr = format!("http://{}:{}", host, port);
+    /// Constructs a `StorageReadServiceClient` with given SocketAddr.
+    pub fn new(address: SocketAddr) -> Self {
+        let http_addr = format! {"http://{}", address.to_string()};
 
         Self {
             client: Mutex::new(None),
-            addr,
+            http_addr,
         }
     }
 
     async fn client(&self) -> Result<StorageClient<tonic::transport::Channel>, tonic::Status> {
         if self.client.lock().unwrap().is_none() {
-            let client = StorageClient::connect(self.addr.clone())
+            let client = StorageClient::connect(self.http_addr.clone())
```

Comment:
maybe pass by reference? unless it needs the copy.

Reply:
Looks like it needs the copy. The same goes for the connect() call in StorageWriteServiceClient.